module_utils/kafka_consumer_lag.py

```diff
@@ -0,0 +1,156 @@
+                current_offsets[topic] = []
+                topics.append(topic)
+            for partition in topic_partition[1]:
+                # partition[0]=> partition number,
+                # partition[1]=> partition current offset
+                current_offsets[topic].append((partition[0], partition[1]))
+
+        # Get last offset for each topic partition coordinated by each broker
+        # Result object is set up also
+        for broker in brokers:
+            # filter only topic consumed by consumer_group
+            topics_partitions_by_broker = filter_by_topic(
+                                cluster.partitions_for_broker(broker.nodeId),
+                                topics)
+            request_topic_partitions = build_offset_request_topics_partitions(
+                                           topics_partitions_by_broker)
+
+            response = self._send(
+                     consumer_coordinator,
+                     _OffsetRequest(AS_CONSUMER, request_topic_partitions),
+                     _OffsetResponse)
+
+            if response:
+                for topics_partitions in response.topics:
+                    topic = topics_partitions[0]
+                    partitions = topics_partitions[1]
+                    if topic not in results:
+                        results[topic] = {}
+                    for partition in partitions:
+                        partition_id = partition[0]
+                        last_offset = partition[3]
+                        current_offset = \
+                            current_offsets[topic][partition_id][1]
+                        lag = last_offset - current_offset
+                        global_lag += lag
+                        # Set up result object
+                        results[topic][partition_id] = {
+                            'current_offset': current_offset,
+                            'last_offset': last_offset,
+                            'lag': last_offset - current_offset
+                        }
+
+        results["global_lag_count"] = global_lag
+        return results
+
+
+def filter_by_topic(topics_partitions, topics):
+    result = []
+    for topic_partition in topics_partitions:
+        if topic_partition.topic in topics:
+            result.append(topic_partition)
+
+    return result
+
+
+def build_offset_request_topics_partitions(topics_partitions):
+    _topics_partitions = {}
+    for topic_partition in topics_partitions:
+        topic = topic_partition.topic
+        partition = topic_partition.partition
+        if topic not in _topics_partitions:
+            _topics_partitions[topic_partition.topic] = []
+        _topics_partitions[topic_partition.topic].\
+            append((partition, LATEST_TIMESTAMP))
+
+    # convert to array for _OffsetRequest Struct
+    request_topics_partitions = []
+    for tp in _topics_partitions.items():
+        request_topics_partitions.append(tp)
+
+    return request_topics_partitions
```

Comment:
I think you can either `return [for tp in _topics_partitions.items()]` or `return list(_topics_partitions.items())` here

Reply:
Done